Jar-based start does not work

See exception:

```
SCHWERWIEGEND: Application class de.michab.scream.ui.ScreamUi failed to launch
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.smack.util.ServiceManager.getApplicationService(ServiceManager.java:74)
	at org.smack.swing.application.Application.create(Application.java:235)
	at org.smack.swing.application.Application$1.run(Application.java:191)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:79)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:484)
	at org.smack.util.ReflectionUtil.createInstanceX(ReflectionUtil.java:464)
	at org.smack.util.ServiceManager.getApplicationService(ServiceManager.java:61)
	... 15 more
Caused by: java.lang.InternalError: de.michab.scream.RuntimeX: 22 : Ein/Ausgabeoperation ist fehlgeschlagen: /de/michab/scream/extensions/basic.s (No such file or directory)
	at de.michab.scream.Scream.addExtensions(Scream.java:408)
	at de.michab.scream.Scream.createTle(Scream.java:268)
	at de.michab.scream.Scream.<clinit>(Scream.java:664)
	at de.michab.scream.ui.ScreamUi.<init>(ScreamUi.java:51)
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:67)
	... 19 more
Caused by: de.michab.scream.RuntimeX: 22 : Ein/Ausgabeoperation ist fehlgeschlagen: /de/michab/scream/extensions/basic.s (No such file or directory)
	at de.michab.scream.RuntimeX.mIoError(RuntimeX.java:478)
	at de.michab.scream.Scream.load(Scream.java:469)
	at de.michab.scream.Scream.load(Scream.java:422)
	at de.michab.scream.Scream$1._executeImpl(Scream.java:508)
	at de.michab.scream.fcos.Procedure.lambda$0(Procedure.java:97)
	at de.michab.scream.pops.Primitives.evalImpl(Primitives.java:433)
	at de.michab.scream.pops.Primitives.lambda$26(Primitives.java:435)
	at de.michab.scream.pops.Primitives.lambda$24(Primitives.java:414)
	at de.michab.scream.util.Continuation.trampoline(Continuation.java:67)
	at de.michab.scream.util.Continuation.toStack(Continuation.java:164)
	at de.michab.scream.Scream.toStack(Scream.java:350)
	at de.michab.scream.Scream.evalImpl(Scream.java:367)
	at de.michab.scream.Scream.load(Scream.java:465)
	at de.michab.scream.Scream.load(Scream.java:434)
	at de.michab.scream.Scream.addExtensions(Scream.java:401)
	... 23 more


```

--- a/pom.xml
+++ b/pom.xml
@@ -82,13 +82,15 @@
             <artifactId>maven-dependency-plugin</artifactId>
             <executions>
               <execution>
-                <phase>install</phase>
+                <phase>package</phase>
                 <goals>
                   <goal>copy-dependencies</goal>
                 </goals>
+                <inherited>false</inherited>
                 <configuration>
                   <outputDirectory>${project.build.directory}/lib</outputDirectory>
                   <includeScope>compile</includeScope>
+                  <useBaseVersion>false</useBaseVersion>
                 </configuration>
               </execution>
             </executions>

--- a/src/main/java/de/michab/scream/util/LoadContext.java
+++ b/src/main/java/de/michab/scream/util/LoadContext.java
@@ -1,8 +1,9 @@
 /*
  * Scream @ https://github.com/michab/dev_smack
  *
- * Copyright © 2022 Michael G. Binz
+ * Copyright © 2022-2023 Michael G. Binz
  */
+
 package de.michab.scream.util;
 
 import java.io.File;
@@ -12,7 +13,18 @@ import java.io.InputStream;
 import java.net.MalformedURLException;
 import java.net.URL;
 
-// https://flylib.com/books/en/1.134.1/jarurlconnection.html
+/**
+ * Support for file loading.  Implements the necessary logic to load files from
+ * the filesystem, inside a jar file or from the internet.  Supports loading of
+ * several files that are rooted to a single source.  That means if the first
+ * file is rooted in a jar file then follow-up files with a relative name are
+ * loaded from the same source.
+ * <p>
+ * See // https://flylib.com/books/en/1.134.1/jarurlconnection.html for the
+ * basic idea.
+ *
+ * @author micbinz
+ */
 public class LoadContext
 {
     private final String _prefix;
@@ -48,8 +60,8 @@ public class LoadContext
         this( nonThrowingUrl( filename ) );
     }
 
-    public LoadContext( URL filename ) {
-
+    public LoadContext( URL filename )
+    {
         if ( FILE_PROTOCOL.equals( filename.getProtocol() ) )
         {
             _prefix = FILE_PROTOCOL + ":";
@@ -100,7 +112,7 @@ public class LoadContext
         return new LoadContext(
                 prev._prefix,
                 f,
-                _isFile );
+                prev._isFile );
     }
 
     public InputStream getStream() throws IOException
@@ -123,5 +135,14 @@ public class LoadContext
     public String toString()
     {
         return toUrl().toExternalForm();
+    }
+
+    /**
+     * @return {@code true} if this represents a file, otherwise
+     * this is an URL.
+     */
+    public boolean isFile()
+    {
+        return _isFile;
     }
 }

--- a/src/test/java/de/michab/scream/util/LoadContextTest.java
+++ b/src/test/java/de/michab/scream/util/LoadContextTest.java
@@ -26,6 +26,7 @@ public class LoadContextTest
                 new File( "/com/elharo/io/StreamCopier.class" ),
                 lc.getFile() );
     }
+
     @Test
     public void urlFile() throws Exception
     {
@@ -62,6 +63,7 @@ public class LoadContextTest
                 "/de/michab/scream/extensions/a.s";
         LoadContext lca = new LoadContext( a );
         assertTrue( lca.isAbsolute() );
+        assertTrue( lca.isFile() );
 
         String b =
                 "b.s";
@@ -69,19 +71,19 @@ public class LoadContextTest
         assertFalse( lcb.isAbsolute() );
 
         LoadContext lcc = lcb.relate( lca );
-        var file = lcc.getFile();
+        assertTrue( lcc.isFile() );
         assertEquals(
                 new File( "/de/michab/scream/extensions/b.s" ),
-                file );
+                lcc.getFile() );
 
         LoadContext lcd = new LoadContext( "/tmp/313.so" );
         assertTrue( lcd.isAbsolute() );
 
         LoadContext lcf = lcd.relate( lcc );
-        file = lcf.getFile();
+        assertTrue( lcf.isFile() );
         assertEquals(
                 new File("/tmp/313.so"),
-                file );
+                lcf.getFile() );
     }
 
     @Test
@@ -91,26 +93,29 @@ public class LoadContextTest
                 "file:/de/michab/scream/extensions/a.s";
         LoadContext lca = new LoadContext( a );
         assertTrue( lca.isAbsolute() );
+        assertTrue( lca.isFile() );
+
         String b =
                 "b.s";
         LoadContext lcb = new LoadContext( b );
         assertFalse( lcb.isAbsolute() );
 
         LoadContext lcc = lcb.relate( lca );
-        var file = lcc.getFile();
-        assertTrue( lca.isAbsolute() );
+        assertTrue( lcc.isAbsolute() );
+        assertTrue( lcc.isFile() );
         assertEquals(
                 new File( "/de/michab/scream/extensions/b.s" ),
-                file );
+                lcc.getFile() );
     }
 
     @Test
-    public void relativeJar() throws Exception
+    public void relativeJarHttp() throws Exception
     {
         URL a = new URL(
                 "jar:http://www.oreilly.com/javaio.jar!/com/elharo/io/A.class");
         LoadContext lca = new LoadContext( a );
         assertTrue( lca.isAbsolute() );
+        assertFalse( lca.isFile() );
 
         String b =
                 "B.class";
@@ -118,12 +123,42 @@ public class LoadContextTest
         assertFalse( lcb.isAbsolute() );
 
         LoadContext lcbr = lcb.relate( lca );
-        var file = lcbr.getFile();
+        assertFalse( lcbr.isFile() );
         assertEquals(
                 new File( "/com/elharo/io/B.class" ),
-                file );
+                lcbr.getFile() );
         assertEquals(
                 "jar:http://www.oreilly.com/javaio.jar!/com/elharo/io/B.class",
                 lcbr.toString() );
     }
+
+    @Test
+    public void relativeJarFile() throws Exception
+    {
+        URL a = new URL(
+                "jar:file:/Users/micbinz/git/github/app_scream/target/app_scream-0.0.1-SNAPSHOT.jar!/de/michab/scream/extensions/common-init.s");
+        LoadContext lca = new LoadContext( a );
+        assertTrue( lca.isAbsolute() );
+        assertFalse( lca.isFile() );
+
+        String b =
+                "basic.s";
+        LoadContext lcb = new LoadContext( b );
+        assertFalse( lcb.isAbsolute() );
+
+        LoadContext lcbr = lcb.relate( lca );
+        assertFalse( lcbr.isFile() );
+        assertEquals(
+                new File( "/de/michab/scream/extensions/basic.s" ),
+                lcbr.getFile() );
+        assertEquals(
+                "jar:file:/Users/micbinz/git/github/app_scream/target/app_scream-0.0.1-SNAPSHOT.jar!/de/michab/scream/extensions/basic.s",
+                lcbr.toString() );
+    }
+
+//    WARNUNG: stack.peek()= isFile=false
+//    Feb. 22, 2023 8:52:17 PM de.michab.scream.Scream load
+//    WARNUNG: current=jar:file:/Users/micbinz/git/github/app_scream/target/app_scream-0.0.1-SNAPSHOT.jar!/de/michab/scream/extensions/basic.s isFile=true
+//    Feb
+
 }

